servicex/servicex_adapter.py

```diff
@@ -133,11 +133,20 @@ async def get_transform_status(self, request_id: str) -> TransformStatus:
         headers = await self._get_authorization()
         retry_options = ExponentialRetry(attempts=5, start_timeout=10)
         async with RetryClient(retry_options=retry_options) as client:
-            async with client.get(url=f"{self.url}/servicex/transformation/{request_id}",
-                                  headers=headers) as r:
-                if r.status == 401:
-                    raise AuthorizationError(f"Not authorized to access serviceX at {self.url}")
-                if r.status == 404:
-                    raise ValueError(f"Transform ID {request_id} not found")
-                o = await r.json()
-                return TransformStatus(**o)
+            retries = 5
```

Comment:
I hate to be obnoxious and ask you to learn a new package, but perhaps you can use `tenacity` (https://tenacity.readthedocs.io/en/latest/ , which Ben has been using in the backend code for retries)? it simplifies a lot of this logic, something like

```
from tenacity import AsyncRetrying, RetryError, stop_after_attempt, wait_fixed

try:
    async for attempt in AsyncRetrying(stop=stop_after_attempt(5), wait=wait_fixed(3), reraise=True):
        with attempt:
            async with client.get( ... )
except RuntimeError as e:
     ....
````

Reply:
(I've been asking for tenacity >= 9.0.0)

Reply:
I agree here - please use a package that has done this - there are some really nice ones out there - and they handle all the logic we could imagine way better than us. `tenacity` is pretty nice!!